zebra-chain/src/block.rs

```diff
@@ -86,6 +87,34 @@ impl Block {
             Some(height) => Commitment::from_bytes(self.header.commitment_bytes, network, height),
         }
     }
+
+    /// Check if the `network_upgrade` field from the transactions match block height.
+    ///
+    /// # Consensus rule:
+    ///
+    ///  The nConsensusBranchId field MUST match the consensus branch ID used for
+    ///  SIGHASH transaction hashes, as specifed in [ZIP-244] ([7.1]).
+    ///
+    /// [ZIP-244]: https://zips.z.cash/zip-0244
+    /// [7.1]: https://zips.z.cash/protocol/nu5.pdf#txnencodingandconsensus
+    pub fn check_transaction_network_upgrades(
```

Comment:
Actually I think @dconnolly was right here.

(I didn't see her original feedback because it had been resolved automatically.)
```suggestion
    pub fn check_transaction_network_upgrade_consistency(
```

Reply:
https://github.com/ZcashFoundation/zebra/pull/2354/commits/5520611d46bea6cfaca662b4528ca0ddba64c9e1